### DISCUSSION TITLE
Reducing non-ping latency

### DISCUSSION THREAD

**Discussion by leoinlios** [jamulussoftware/jamulus#2075] [Technical Support & Help]
Hi guys - here is the setup - THANK YOU for your help:

Roland TD-27 KV drum kit module ->
USB cable ->
USB 3.0 motherboard port of DESKTOP gaming Win 10 PC (16GB ram - SSD - i7 6700 processor @3.4GHz - 4GB GPU), latest Windows Updates ->
WIRED ethernet broadband at 38 Mbps down, 6 Mbps up -> TD-27 ASIO driver 1.0 ->
Jamulus 3.8.1 rc1 (Using London servers)

Here are my ASIO/Jamulus settings:
![image_38490](https://user-images.githubusercontent.com/4964741/137877866-5ccce603-0a87-49fc-bd97-62ea8a016cbb.jpg)

For added info, when I use my DAW (Cakewalk) to record my e-drums, using the exact same setup as above, the DAW reports:
![image_38491](https://user-images.githubusercontent.com/4964741/137877976-a13b7a1f-d695-4064-a0c7-7c5c7dc68b79.jpg)

I am new to e-drums/pc connectivity. I have read a bit but feel a bit overwhelmed by the complexity and also multiple recommendations, please have me patience. My issue is this: I am told that 10ms ping is good but 30ms delay is high and indeed other musicians have pointed this out to me, though when I hear my own feed, it is marginally noticeable. So...

How can this be improved?

1) Invest in better PC hardware (replace mobo, ram, ssd, gpu, etc to more recent spec)?
2) Invest in a PCI-e sound card so I don't use the onboard usb port / onboard processing?
3) Invest in an external Interface? (surely anything in between the td27 and the pc can only ADD latency, no?)
4) Install Linux and run Jamulus there instead?

Thank you so much for your time and advice.

**Comment by henkdegroot** [comment]
Just my opinion, but I don't think there much improvement possible here. Also 30ms is not really bad, but indeed depending on the type of music and instrument it might be a different experienc. With our band (20-25 players) most of the people have 40ms delay. I have seen rare moments that my delay came down to 25ms but on average it was between 30-40ms and did not cause any issues. My win10 laptop has a much lower spec compared to yours. I do admit, I am using an external audio device which comes with its own asio driver and use microphone to pick up the sound from my instrument. I wonder is using the "sound" output of the drums through an audio device can improve the delay, but knowing my experience it is most likely not going to be much lower than 30ms.

No much help in the above, but still believe worth sharing.

**Reply by gilgongo** [reply]
While Henk is right about you being close to minimal latency, I think I'm right in saying the TD-27 doesn't let you turn off monitoring. Can you hear yourself when you are **not** connected to a server? If so, this will mean that even with a realtively small delay, you will start dragging behind other musicians - the latency will feel (to them and you) worse than it really is. 

The solution in this case would be to plug the TD-27 into an audio interface that allows you to turn off monitoring (I use a UMC202).

**Reply by leoinlios** [reply]
hi, thanks @henkdegroot . 

@gilgongo "the TD-27 doesn't let you turn off monitoring" - I think that's right. You always hear yourself when playing (from the TD-27 module, and optionally, from your Jamulus sound coming from the server.). I can mute my Jamulus sound, but I cannot mute my coming out of the phones from my module. This is why I can only play by listening to myself from the module and everybody else from Jamulus, i.e. not me from Jamulus. If there was a way to just listen to myself from Jamulus, it would be madness anyway as I would be hitting pads and moments later hearing the sound, which as a drummer that is total insanity and you cannot play anything serious or tight. 

Please can you explain about the use of an external interface? How can a device in between my module and my PC ever *reduce* latency? Also, how would it be connected? The interface you mentioned has a single USB connector, surely that is to and from the PC, if so, where/how do I connect my drum module, which uses also USB in/out too? Do I need an interface with 2 USB ports then (one for the module and one for the PC)? 

Sorry, lots of questions. Thanks

**Reply by gilgongo** [reply]
On interfaces: You are currently using the TD-27's audio interface (and ASIO driver on your PC for that).  So instead you would take the analogue audio output from the TD-27 and put that into the analogue input of the audio interface. No additional latency is involved. 

On latency and listening to your own signal: The way Jamulus works is that everyone sends their sound to the server, which mixes it then sends that mix out to each person (or "channel" in the jargon). This means that if you are listening to your own sound _before_ it gets to the server then what you play is (assuming some latency) guaranteed to be ahead of the other musicians as you hear them. And you will likely want to slow down to compensate for their apparent lack of attention to your time keeping :-) But that will be futile as you will eventually grind to a halt.

This is why you are strongly encouraged to [only listen to the sound from the server](https://jamulus.io/wiki/Client-Troubleshooting#you-all-sound-ok-but-its-difficult-to-keep-together). At least that way, you are just dealing with a reasonably constant latency and can compensate here and there quite easily.

> which as a drummer that is total insanity and you cannot play anything serious or tight.

That will come as a surprise to the numerous drummers who use Jamulus all the time on delays longer than 30ms. :-) I'm a drummer, and can usually deal with up to 45ms and even a bit beyond depending on the genre. BTW 28ms is the equivalent to playing with somebody/speakers about 10m away in the real world.

**Reply by leoinlios** [reply]
Very interesting, thanks for taking the time. So:

TD-27 master analogue output ----> External interface analogue input
External interface USB output ----> PC USB port
Jamulus input is set to interface output
Jamulus output is set to interface input
External interface phones output ----> Headphones

If the setup above is correct, it means the interface has 2 inputs, 1) my TD-27 analogue, 2) Jamulus entire server mix output via USB. In line with the advice to listen to your sound from the server (not your instrument), I should therefore mute input 1 (my original drum sounds). Is this possible and did I get this all correct?

Re the playing by listening to your sound AFTER you strike a pad, we will have to agree to disagree. I play fusion and some upbeat and accentuated music...I find it chaotic in my brain to hear me with a delay. Strike - gap - sound.  The  only solution I have resorted to is to play at my own fixed tempo and, as you said, keep going while the rest are "off" my tempo a bit. At least I know they hear me on tempo and the chaos is slightly less. :)

**Reply by gilgongo** [reply]
That setup would work, but depending on the model of the audio interface you should have some way of turning off monitoring (on my UMC202 there is a toggle switch marked "Direct monitor") so you only hear the sound that's coming _into_ the interface from the PC, not _going out_ of it to the PC, as you have today. That might be referred to as "muting" perhaps, not sure.

Some drummers (and other musicians) do seem to find it hard to deal with even small latencies. But even if you find it easy, listening to yourself in the band's mix is pretty much essential, then playing consistenly to your own "clock" so that others (who, BTW will be dealing with their own latencies slightly different from yours) will lock into you. 

I would also point out that while good timing is of course desireable for performances and recording, Jamulus is mainly about jamming and rehearsing, where keeping it in pocket isn't so important. I also quite like a bit of chaos in rehearsing becuase it make me feel more confident that I could handle similar in performances. I play with backing tracks and sequencers and stuff that can go wrong in various ways on stage. While they might not generate unexpected latency, it's nice to have some "resiliance training" when playing. A bit like when your ride stand collapses or your bass drum pedal snaps, for example. The show must go on :-)

**Reply by leoinlios** [reply]
haha nice. Yes, one way or another I get chaos from playing this way and it is good training as you have alluded. 

Will check interfaces to see if that improves the latency. (Also good as I could add a Mic which most musicians have and I do not). 

Thanks for your help
